### PR TITLE
Wip/update 3rd party ws

### PR DESCRIPTION
**Pull request by lrosenth**
@tobiasschweizer @subotic Modified SIPI to use newest kakadu, PNG etc. libraries. Fixed a bug with paramers for POST requests with multipart/form-data (thanks @Vijeinath)

@subotic: Could You add newest kakadu to travis build...

### New features:

- Added latest kakadu version v7_A_4-01727L.zip
- support for CIELab for both 8- and 16-bit images
- try/catch for ICC profiles that are not supported by kakadu. These profiles are added to the "essential metadata" in order to be reinstated if the JPX is converted back to a TIFF or JPEG.
- added more unit tests

### Bugfixes

- 16 Bit PNG images are now teated correctly by byteswapping of the data (htons), since PNG uses network byte order which is usually noit zthe host byte order on intel processors
- alpha channels are treated correctly with JPEG2000
- The parameter names in a multidata/form-data POST request now have the double quotes removed

### CODE REVIEW

**Comment by tobiasschweizer** [comment]
@lrosenth Could you adapt the kakadu version in the docs (Prerequisites, https://github.com/dhlab-basel/Sipi/blob/a16094978ecbaa489bfbc4340ab89b9f5d3f2d06/manual/building.rst)?

**Comment by subotic** [comment]
I see that there are a number of libraries updated which have links to Sourceforge. Since we don't use Sourceforge anymore (#227), I will upload them to Bintray.

File: config/sipi.config.lua [hunk 1]
```diff
@@ -191,7 +191,7 @@ sipi = {
     --
     -- loglevel, one of "EMERGENCY", "ALERT", "CRITICAL", "ERROR", "WARNING", "NOTICE", "INFORMATIONAL", "DEBUG"
     --
-    loglevel = "ERROR",
+    loglevel = "DEBUG",
```
Comment (subotic) [inline config/sipi.config.lua]: Should be changed back to `ERROR`.

File: config/sipi.init.lua [hunk 2]
```diff
@@ -230,16 +230,47 @@ end
 --    filepath: server-path where the master file is located
 -------------------------------------------------------------------------------
 function pre_flight(prefix,identifier,cookie)
+    local success, result
+   if prefix == 'imgrep' then
+        local url = 'http://127.0.0.1/salsah/api/iiif/' .. identifier
```
Comment (subotic) [inline config/sipi.init.lua]: Please no hardcoded IPs. Make this configurable.

**Comment by tobiasschweizer** [comment]
@lrosenth I can review this PR next week. I would also like to test it with the 3000 image regions that create a huge load on the server.

Also the Knora integration tests should be executed.

@subotic Can we make the Knora integration tests heavier as discussed before?

**Comment by subotic** [comment]
> Can we make the Knora integration tests heavier as discussed before?

Yes, but probably not before next week. I will try to whip up something.

**Comment by tobiasschweizer** [comment]
@subotic I will work on it next week at the earliest

**Comment by tobiasschweizer** [comment]
@lrosenth By when do you need the review for this PR?

File: .travis.yml [hunk 3]
```diff
@@ -29,14 +29,17 @@ jobs:
     - stage: test
       script:
         # decrypt archive containing the kakadu library
-        - openssl aes-256-cbc -K $encrypted_53935862068b_key -iv $encrypted_53935862068b_iv -in vendor/v7_A_2-01382N.zip.enc -out vendor/v7_A_2-01382N.zip -d
+        - openssl aes-256-cbc -K $encrypted_53935862068b_key -iv $encrypted_53935862068b_iv -in vendor/v7_A_3-01382N.zip.enc -out vendor/v7_A_3-01382N.zip -d
```
Comment (tobiasschweizer) [inline .travis.yml]: @subotic I think this should be `v7_A_4-01727L.zip.enc`, right?

**Comment by tobiasschweizer** [comment]
Locally it compiles, but the some of tests fail (probably a configuration issue):

```
$ make test
Running tests...
Test project Sipi_github/build
    Start 1: all_python_based_e2e_tests
1/3 Test #1: all_python_based_e2e_tests .......   Passed   33.46 sec
    Start 2: configuration_unit_test
2/3 Test #2: configuration_unit_test ..........***Failed    0.05 sec
    Start 3: sipiimage_unit_test
3/3 Test #3: sipiimage_unit_test ..............***Failed    0.10 sec

33% tests passed, 2 tests failed out of 3

Total Test time (real) =  33.62 sec

The following tests FAILED:
	  2 - configuration_unit_test (Failed)
	  3 - sipiimage_unit_test (Failed)
Errors while running CTest
make: *** [test] Error 8
```

File: .travis.yml (hunk 3 above)
Comment (tobiasschweizer) [reply .travis.yml]: https://github.com/dhlab-basel/Sipi/pull/229/commits/1551367c56312ee7012d8806e1ea69a7e595a31b thanks!

Comment (subotic) [reply .travis.yml]: your welcome :-)

**Comment by subotic** [comment]
Sipi tests are now fixed. One of them is not passing and ends in a `SegFault`.

```
3: Test command: /sipi/build/test/unit/sipiimage/sipiimage
3: Test timeout computed to be: 9.99988e+06
3: [==========] Running 3 tests from 1 test case.
3: [----------] Global test environment set-up.
3: [----------] 3 tests from Sipiimage
3: [ RUN      ] Sipiimage.CheckIfTestImagesCanBeFound
3: [       OK ] Sipiimage.CheckIfTestImagesCanBeFound (0 ms)
3: [ RUN      ] Sipiimage.ConvertTiffWithAlphaToJPG
3/3 Test #3: sipiimage_unit_test ..............***Exception: SegFault  0.22 sec
```

**Comment by subotic** [comment]
Tests are failing. You can run them locally with: `ctest --verbose`

**Comment by lrosenth** [comment]
@tobiasschweizer @subotic 
Please review – would like to merge!

**Comment by subotic** [comment]
@lrosenth I have merged this branch into `develop`. I will make a new release today.

**Pull request merged**